comment/tests/test_service.py

```diff
@@ -30,10 +29,11 @@ def setUp(self):
     def setUpTestData(cls):
         super().setUpTestData()
         cls.child_comment = cls.create_comment(cls.content_object_1, parent=cls.comment_1)
+        cls.email_service = DABEmailService(cls.comment_obj, cls.request)
 
     def email_contents_test(self, contents, *args):
         for arg in args:
-            self.assertTrue(arg in contents)
+            self.assertIs(arg in contents, True, msg=f'{arg} not present in email contents')
```

Comment:
What the purpose of using this form of assertion? This might be the issue since I have never encountered this fails with the old assertion `self.assertTrue(arg in contents)`.
Also where `msg=f'{arg} not present in email contents'` come from? I mean is it important info for our test?

Reply:
It still doesn't work. You may check this at your end too by cloning this branch and try to run the command `watch -n 3 -d python mange.py test --keepdb comment.tests.test_service`.

> What the purpose of using this form of assertion?

This form of assertion is more explicit in this case. If you think both of them work differently, you may look up at their source [assertTrue](https://github.com/python/cpython/blob/master/Lib/unittest/case.py#L677) and [assertIs](https://github.com/python/cpython/blob/master/Lib/unittest/case.py#L1115).

> This might be the issue since I have never encountered this fails with the old assertion self.assertTrue(arg in contents)

You're right, the tests on`develop` branch are working as expected. Maybe it has got to do something with the change of `EmailService` from `setUp` method to `setUpClass` method in [`TestDABEmailService`](https://github.com/Radi85/Comment/pull/189/files#diff-f2e7d9c153f774cd79acae65f97ea94af7b74f04760f3d9e37dec05bbfec5ccaR17) or some change in the initialization of [`BaseAnonymousCommentTest`](https://github.com/Radi85/Comment/pull/189/files#diff-c218deacbe7b0c004243620d56d658670d0cb36bb5aa08a6d1bc19f6909761b7R123)or something else.

Things are pretty messed up here due to the pandemic but will try to see today if I can figure this out to distract myself.

> Also where msg=f'{arg} not present in email contents' come from? I mean is it important info for our test?

Since this test is running inside a loop, this message tells us about the argument, if and when the test fails. Otherwise, it just takes a couple of more steps to figure out exactly what caused the failure.

Reply:
Since the issue doesn't popup with the `assertTrue`, I would suggest to use it again to avoid investing more time with this issue.

Reply:
does the issue not popup after the change at your end on my branch? it does on my end as i mentioned in my  previous comment https://github.com/Radi85/Comment/pull/189#discussion_r614574736, there are some other possible variables that could potentially cause this.

Reply:
wow, i am pleased(i'm not sure i should be) to announce that it does actually fail on the `develop` branch as well. i'm pasting a script below that you too may use to test in your local machine. In my case, it actually failed after 62 iterations.

```bash
#!/usr/bin/bash

iteration=0
while true; do
    iteration=$((iteration + 1))
    #pytest comment/tests/test_service.py::TestDABEmailService::test_send_confirmation_request_django --pdb
    python manage.py test --keepdb comment.tests.test_service.TestDABEmailService.test_send_confirmation_request_django
    if [[ $? -ne 0 ]]; then
        echo -e "-----------\nit fucking failed in the $iteration iteration--------------------\n"
        break
    fi
    echo -e "-------------\nrunning the $iteration iteration\n-----------------\n"
done

```

Reply:
I am not sure what the purpose of running a test infinit times to see if fails or not 🤔 .

I personally cannot see where the issue is and as long as the test is passing on travis I would accept it.

I might look into this tomorrow but I don't promise 😄

Reply:
> I am not sure what the purpose of running a test infinit times to see if fails or not thinking .

actually it does fail sometimes, but not always. out of the blue, it failed for me and that is how i noticed it. it is quite hard to reproduce because it passes more often than not.  i just wanted to confirm that it fails, without having to use visual feedback.

> I personally cannot see where the issue is and as long as the test is passing on travis I would accept it.

the test is flaky, that's the issue at the moment. i too am yet to figure out the exact reason for this.